### PR TITLE
azure: update docs for explicit anonymous login

### PR DESCRIPTION
**Pull request by isidentical**
Per https://github.com/iterative/dvc/pull/5833

### CODE REVIEW

**Comment by isidentical** [comment]
As I linked in the original issue, this is per iterative/dvc#5833

**Comment by jorgeorpinel** [comment]
OK thanks @isidentical. Please try to link from the docs PRs too. Back links from other tickets aren't a clear indication and get easily lost in all the PR activity.

**Comment by jorgeorpinel** [comment]
(I updated the description above.)

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [inline content/docs/command-reference/remote/modify.md:343]:
```suggestion
3. DVC will try next to connect with `account_key` or `sas_token` (in this
   order) if either are provided.
```

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [inline content/docs/command-reference/remote/modify.md:345]: Unlike the other things in this list, this doesn't appear to be a "step." It's just a config flag. Maybe move to a note below this list?

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [inline content/docs/command-reference/remote/modify.md:347]:
```suggestion
5. If none of the steps above apply, and the only variable that is set is
   `account_name`, then DVC will try to use a
```

I think @jorgeorpinel would prefer "that."

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [inline content/docs/command-reference/remote/modify.md:350]: "activated logings" looks like a typo but idk what was intended.

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [inline content/docs/command-reference/remote/modify.md:364]:
```suggestion
- `account_name` - storage account name. Required if `connection_string` is not
```

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [inline content/docs/command-reference/remote/modify.md:365]: "Will accompany the options" - not sure what this means. Is it an instruction ("must accompany")? An option ("may accompany")? A request ("should accompany")? Something else?

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [inline content/docs/command-reference/remote/modify.md:368]: not really sure about naming/style convention here:

- `myremote`/`my_remote`/`'myremote'`
- `accountname`/`account_name`/`'accountname'`
- `myuser`/`my_user`/`'myuser'`

And of course `'my_remote'`/`'account_name'`/`'my_user'`. Is there some important meaning behind this combination?

Same issue with all commands below.

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [inline content/docs/command-reference/remote/modify.md:406]:
```suggestion
* `allow_anonymous_login` - whether to fallback to anonymous login. If none
  of the options above are specified besides `account_name`, DVC will try
  to connect anonymously for read-only buckets:
```

**Review by casperdcl — DISMISSED** [review]

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:345]: Ah these aren't steps @casperdcl, the list reflects the order in which auth params are used by DVC (Azure remotes). So if neither 1, 2, or 3 are set, this one is checked. I'd just add a reference to what anonymously is:

```suggestion
4. If `allow_anonymous_login` is set to `True`, then DVC will try to connect
   [anonymously]((https://docs.microsoft.com/en-us/azure/storage/blobs/anonymous-read-access-configure)).
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:338]: > If `connection_string` is not set, the `account_name` must be given.

@isidentical do `tenant_id`, `client_id`, `client_secret` require `account_name`? That's kind of what it looks like from the change above. Can you explain so we can take this over? Thanks

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:343]: Thanks, good copy edit. We can usually just commit these BTW (we don't expect the core team to write perfectly). BUT I have a Q on this (https://github.com/iterative/dvc.org/pull/2489#pullrequestreview-665102806) so let's wait a bit...

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:347]: Or neither ("the only variable set"). Wouldn't call these steps though. Updating and committing along with any other copy edits...

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:350]: It's just TMI in the parenthesis, the user can read the details in the link, I think. Let's simplify?

Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:350]: Also, looks like the paragraph before the list needs an update. Could move some of that language into this point.

> By default, DVC authenticates using an Azure default credential (if any). This uses certain env...

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:365]: This relates to my Q in https://github.com/iterative/dvc.org/pull/2489#pullrequestreview-665102806. I think it means `account_name` is almost always required now but let's see what @isidentical says.

Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:365]: BTW we seem to be missing a bullet entry for `connection_string` ???

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:368]: Good Qs but unrelated to this PR (specific DVC update)

But to anser, `account_name` comes from DVC. The other ones are just sample values. Currently we don't hyphenate `'myxyz'`, so this seems consistent.

Resolving (feel free to reopen if I didn't understand you).

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:392]: Should mention about default credentials here too instead anonymous auth?

> p.s. I rescued the anon auth link for step 4 above.

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:395]: Reverting (no need to change this)
```suggestion
  $ dvc remote modify --local myremote account_key 'mysecret'
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:401]: Same
```suggestion
  $ dvc remote modify --local myremote sas_token 'mysecret'
```

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:406]: Nice. Can probably simplify even more? E.g. "Only has an effect along with `account_name`."

Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:406]: read-only -> public?

**Comment by jorgeorpinel** [comment]
@isidentical lots of notifications probably, please check directly on GH as many reviews are already resolved. Also I see the core PR is still ongoing so maybe we'll check it out (and possibly chime in there) first, to better understand this change.

For now I labeled this `awaiting-dvc-merge` (please remind us when that one gets merged if needed).

File: content/docs/command-reference/remote/modify.md
Comment (isidentical) [reply content/docs/command-reference/remote/modify.md:338]: Yes, they do. Basically, you should either have `connection_string` or `account_name` set all the time; https://github.com/dask/adlfs/blob/96958331854c5a67917d4c9a802a62720d091374/adlfs/spec.py#L493-L526

File: content/docs/command-reference/remote/modify.md
Comment (isidentical) [reply content/docs/command-reference/remote/modify.md:406]: > read-only -> public?

yes

File: content/docs/command-reference/remote/modify.md
Comment (isidentical) [reply content/docs/command-reference/remote/modify.md:365]: > I think it means account_name is almost always required now but let's see what @isidentical says.

Yes, that is the case.

Comment (isidentical) [reply content/docs/command-reference/remote/modify.md:365]: > BTW we seem to be missing a bullet entry for connection_string ?

This bullet partially covers it, but if you'd like to add it as a separate one, sure!

Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:365]: >  you'd like to add it as a separate one, sure

Please 🙂

File: content/docs/command-reference/remote/modify.md
Comment (isidentical) [reply content/docs/command-reference/remote/modify.md:392]: We already mention it in the step 5, so I don't think that is necessary.

**Comment by jorgeorpinel** [comment]
Ping @casperdcl 🙂

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [reply content/docs/command-reference/remote/modify.md:343]: that comment is marked as resolved?

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [reply content/docs/command-reference/remote/modify.md:345]: > the list reflects the order in which auth params are used

exactly.

- `allow_anonymous_login` is not in my mind an "auth param" but rather an "auth modifier option" so is out of place in this list
- is it really checked only at the fourth step (i.e. ignored if steps 1-3 succeed)?

The list may be correct (I haven't reviewed the underlying code to check) but I doubt it.

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [reply content/docs/command-reference/remote/modify.md:365]: > almost always required

not at all clear. If it is merely "almost" then we shouldn't open with a statement which strongly implies "absolutely always" and then go on to contradict ourselves later by saying that it's optional.

I'd recommend leaving this out

```suggestion
  supplied.
```

"Will accompany the options below" sounds a bit like "works in tandem with the options below" which in turn sounds like "CLI options affect each other" which in turn sounds like either redundant info or unneeded implementation detail.

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [reply content/docs/command-reference/remote/modify.md:406]: > read-only -> public

updated comment.

> Only has an effect along with

doesn't have the same meaning as the (admittedly clunky) phrase

> none of the options above are specified besides

**Comment by casperdcl** [comment]
didn't think my response was needed - was still waiting for things to be addressed but in any case have responded where possible.

File: content/docs/command-reference/remote/modify.md
Comment (isidentical) [reply content/docs/command-reference/remote/modify.md:345]: > is it really checked only at the fourth step (i.e. ignored if steps 1-3 succeed)?

yes.

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [inline content/docs/command-reference/remote/modify.md:333]: I updated this wording btw, hope it's ok.

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:343]: > that comment is marked as resolved?

Yeah, @isidentical answered the Q so he resolved it. I unresolved it now for visibility.

Anyway, looks like we can just commit this...

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:338]: OK thanks @isidentical but it would still be great if you can elaborate because the doc says "DVC authenticates using an Azure default credential" first, and then the list says that either `connection_string` or `account_name` must be given. Which one is it? Thanks

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:333]: Actually @casperdcl, the rewording here changed the semantics. These params are only used **if set** (a default credential is the used otherwise, as indicated at the beginning of the paragraph). Not 100% sure though, please see https://github.com/iterative/dvc.org/pull/2489#discussion_r643676441.

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:365]: Up to you here @casperdcl, as long as things are clear. I agree to remove this sentence but maybe there's some way to indicate in each param which other params it works with?

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: > allow_anonymous_login is not in my mind an "auth param"

I see what you meant. We call all of these (config) params. They're params that can be sent `remote modify` (for Azure remotes)

File: content/docs/command-reference/remote/modify.md
Comment (isidentical) [reply content/docs/command-reference/remote/modify.md:338]: > Which one is it? Thanks

In either case (even with default credentials) the `account_name` should be given. Previously whether the user wanted anonymous login or not was a bit ambiguous since both default credentials and anonymous login wanted it so we tried to infer it in various terms but now it is done by an explicit option so if you only specify `account_name` then you will use default credentials but if you also specify `allow_anonymous_login` then you will use anonymous login.

**Review by jorgeorpinel — COMMENTED** [review]
Sorry for the delay.

Yes well it looks like we have almost all the info we need from @isidentical (pending a final resolution on the Q of https://github.com/iterative/dvc.org/pull/2489#pullrequestreview-665102806). Mind taking it over @casperdcl ?

Thanks guys

**Comment by jorgeorpinel** [comment]
p.s. I resolved everything that no longer seemed relevant.

File: content/docs/command-reference/remote/modify.md
Comment (casperdcl) [reply content/docs/command-reference/remote/modify.md:365]: maybe a table?

Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md:365]: Sure we have a few e.g. https://dvc.org/doc/user-guide/project-structure/pipelines-files#stage-entries

**Comment by jorgeorpinel** [comment]
Ping @casperdcl could you communicate the situation here? Thanks

**Comment by casperdcl** [comment]
Er it looks like you have 4 comments pending your own suggestions and resolution, @jorgeorpinel

**Comment by jorgeorpinel** [comment]
@casperdcl sorry for the confusion. I just checked the reviews above and I'm not seeing any committable suggestions, and most of them were opened by you, I think.

- In your https://github.com/iterative/dvc.org/pull/2489#discussion_r636558915 I recommended you simplify the whole passage.
- In https://github.com/iterative/dvc.org/pull/2489#discussion_r644000923 you proposed making a table and I agreed.
- https://github.com/iterative/dvc.org/pull/2489#pullrequestreview-665102806 was just a Q to make sure we all understand the changes correctly (no direct action needed).
- In https://github.com/iterative/dvc.org/pull/2489#pullrequestreview-669104124 I mentioned that your change seems incomplete.

Will you be taking this one over? Thanks

**Comment by jorgeorpinel** [comment]
p.s. I accidentally assigned myself along with you and then immediately unassigned myself. Maybe that was the confusion, sorry! Fat fingers 😬 

image

**Comment by jorgeorpinel** [comment]
Ping

**Comment by jorgeorpinel** [comment]
> Taking this over

(as agreed in docs meeting)

File: content/docs/command-reference/remote/modify.md
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: Fixed in 843ff79.

**Comment by jorgeorpinel** [comment]
@isidentical sorry for the long delay. I think I was able to clarify this. Can you do a last check? I'll merge after that. Thanks!

**Review by jorgeorpinel — APPROVED** [review]
OK well I'm going to merge this. Please lmk if you notice anything wrong.

**Pull request merged**
